Source/core/MessageException.h

```diff
@@ -21,23 +21,42 @@
 #define EXCEPTION_H
 
 #include <exception> // For exception class
+#include <errno.h>
 
 #include "Module.h"
 #include "Portability.h"
 #include "TextFragment.h"
+#include "Serialization.h"
 
 namespace Thunder {
 namespace Core {
 
-    class MessageException : public std::exception {
+    class  MessageException : public std::exception {
     private:
-        MessageException();
+        inline MessageException()
```

Comment:
According to the latest guidelines, we need to declare at least the move and the copy constructor. so please declare them = deleted, also think/hope that the default constructor is not needed, so set it as deleted as well and all public.
Also, since the second parameter in the constructor has a default, make sure that single parameter constructors are made explicit (see example I shared).

Reply:
Now, you mention it. I remember vaguely such comment from one of the discussions, but not the documentation. It will be sorted.